From commits-return-16096-apmail-openoffice-commits-archive=openoffice.apache.org@openoffice.apache.org  Wed Nov 14 23:35:16 2012
Return-Path: <commits-return-16096-apmail-openoffice-commits-archive=openoffice.apache.org@openoffice.apache.org>
X-Original-To: apmail-openoffice-commits-archive@www.apache.org
Delivered-To: apmail-openoffice-commits-archive@www.apache.org
Received: from mail.apache.org (hermes.apache.org [140.211.11.3])
	by minotaur.apache.org (Postfix) with SMTP id 82AB7DB39
	for <apmail-openoffice-commits-archive@www.apache.org>; Wed, 14 Nov 2012 23:35:16 +0000 (UTC)
Received: (qmail 3469 invoked by uid 500); 14 Nov 2012 23:35:16 -0000
Delivered-To: apmail-openoffice-commits-archive@openoffice.apache.org
Received: (qmail 3429 invoked by uid 500); 14 Nov 2012 23:35:16 -0000
Mailing-List: contact commits-help@openoffice.apache.org; run by ezmlm
Precedence: bulk
List-Help: <mailto:commits-help@openoffice.apache.org>
List-Unsubscribe: <mailto:commits-unsubscribe@openoffice.apache.org>
List-Post: <mailto:commits@openoffice.apache.org>
List-Id: <commits.openoffice.apache.org>
Reply-To: dev@openoffice.apache.org
Delivered-To: mailing list commits@openoffice.apache.org
Received: (qmail 3422 invoked by uid 500); 14 Nov 2012 23:35:16 -0000
Delivered-To: apmail-incubator-ooo-commits@incubator.apache.org
Received: (qmail 3418 invoked by uid 99); 14 Nov 2012 23:35:16 -0000
Received: from nike.apache.org (HELO nike.apache.org) (192.87.106.230)
    by apache.org (qpsmtpd/0.29) with ESMTP; Wed, 14 Nov 2012 23:35:16 +0000
X-ASF-Spam-Status: No, hits=-2000.0 required=5.0
	tests=ALL_TRUSTED
X-Spam-Check-By: apache.org
Received: from [140.211.11.4] (HELO eris.apache.org) (140.211.11.4)
    by apache.org (qpsmtpd/0.29) with ESMTP; Wed, 14 Nov 2012 23:35:12 +0000
Received: from eris.apache.org (localhost [127.0.0.1])
	by eris.apache.org (Postfix) with ESMTP id D27092388906;
	Wed, 14 Nov 2012 23:34:50 +0000 (UTC)
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: svn commit: r1409557 - in /incubator/ooo/branches/gbuild:
 ext_libraries/serf/win/ main/sfx2/ main/solenv/gbuild/
 main/solenv/gbuild/platform/ main/sw/ main/tools/
Date: Wed, 14 Nov 2012 23:34:49 -0000
To: ooo-commits@incubator.apache.org
From: arist@apache.org
X-Mailer: svnmailer-1.0.8-patched
Message-Id: <20121114233450.D27092388906@eris.apache.org>
X-Virus-Checked: Checked by ClamAV on apache.org

Author: arist
Date: Wed Nov 14 23:34:46 2012
New Revision: 1409557

URL: http://svn.apache.org/viewvc?rev=1409557&view=rev
Log:
gnumake4_060_5ed23c4158ac.patch
# HG changeset patch
# User Michael Stahl <mst@openoffice.org>
# Date 1302528217 0
# Node ID 5ed23c4158ac81f51841e4402547282bf86c3ee9
# Parent  8801515221cfb7634c0262f30bc80b23635cfba0
gnumake4: #i116959#: user defined LDFLAGS:
 rename target local variable LDFLAGS.

Modified:
    incubator/ooo/branches/gbuild/ext_libraries/serf/win/Library_serf.mk
    incubator/ooo/branches/gbuild/main/sfx2/CppunitTest_sfx2_metadatable.mk
    incubator/ooo/branches/gbuild/main/sfx2/Library_qstart.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/CppunitTest.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/Executable.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/Library.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/LinkTarget.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/StaticLibrary.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/linux.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/macosx.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/solaris.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/windows.mk
    incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/winmingw.mk
    incubator/ooo/branches/gbuild/main/sw/Library_vbaswobj.mk
    incubator/ooo/branches/gbuild/main/tools/Library_tl.mk

Modified: incubator/ooo/branches/gbuild/ext_libraries/serf/win/Library_serf.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/ext_libraries/serf/win/Library_serf.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/ext_libraries/serf/win/Library_serf.mk (original)
+++ incubator/ooo/branches/gbuild/ext_libraries/serf/win/Library_serf.mk Wed Nov 14 23:34:46 2012
@@ -44,8 +44,7 @@ $(FIXED_DEF_FILE) : $(ORIGINAL_DEF_FILE)
 	cp $< $@
 	echo serf_bucket_request_set_CL >> $@
 
-$(eval $(call gb_Library_set_ldflags,serf,\
-	$$(LDFLAGS) \
+$(eval $(call gb_Library_add_ldflags,serf,\
 	-DEF:$(FIXED_DEF_FILE)	\
 ))
 

Modified: incubator/ooo/branches/gbuild/main/sfx2/CppunitTest_sfx2_metadatable.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/sfx2/CppunitTest_sfx2_metadatable.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/sfx2/CppunitTest_sfx2_metadatable.mk (original)
+++ incubator/ooo/branches/gbuild/main/sfx2/CppunitTest_sfx2_metadatable.mk Wed Nov 14 23:34:46 2012
@@ -39,9 +39,4 @@ $(eval $(call gb_CppunitTest_add_linked_
     $(gb_STDLIBS) \
 ))
 
-$(eval $(call gb_CppunitTest_set_ldflags,sfx2_metadatable,\
-    $$(LDFLAGS) \
-    -lcppunit \
-))
-
 # vim: set noet sw=4 ts=4:

Modified: incubator/ooo/branches/gbuild/main/sfx2/Library_qstart.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/sfx2/Library_qstart.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/sfx2/Library_qstart.mk (original)
+++ incubator/ooo/branches/gbuild/main/sfx2/Library_qstart.mk Wed Nov 14 23:34:46 2012
@@ -48,12 +48,11 @@ $(eval $(call gb_Library_add_cflags,qsta
     $(filter-out -I%,$(GTK_CFLAGS)) \
 ))
 
-$(eval $(call gb_Library_set_ldflags,qstart_gtk,\
-    $$(LDFLAGS) \
+$(eval $(call gb_Library_add_libs,qstart_gtk,\
 ))
 
-$(eval $(call gb_Library_add_external_libs,qstart_gtk,	\
-    $(patsubst -l%,%, $(filter -l%,	$(GTK_LIBS)))	\
+$(eval $(call gb_Library_add_libs,qstart_gtk,\
+    $(GTK_LIBS) \
 ))
 
 

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/CppunitTest.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/CppunitTest.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/CppunitTest.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/CppunitTest.mk Wed Nov 14 23:34:46 2012
@@ -90,7 +90,9 @@ $(eval $(foreach method,\
 	add_defs \
 	set_defs \
 	set_include \
+	add_ldflags \
 	set_ldflags \
+	add_libs \
 	set_library_path_flags \
 	add_api \
 	add_linked_libs \

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/Executable.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/Executable.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/Executable.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/Executable.mk Wed Nov 14 23:34:46 2012
@@ -77,7 +77,9 @@ $(eval $(foreach method,\
 	add_defs \
 	set_defs \
 	set_include \
+	add_ldflags \
 	set_ldflags \
+	add_libs \
 	set_library_path_flags \
 	add_api \
 	add_linked_libs \

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/Library.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/Library.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/Library.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/Library.mk Wed Nov 14 23:34:46 2012
@@ -120,7 +120,9 @@ $(eval $(foreach method,\
 	add_defs \
 	set_defs \
 	set_include \
+	add_ldflags \
 	set_ldflags \
+	add_libs \
 	set_library_path_flags \
 	add_api \
 	add_linked_libs \

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/LinkTarget.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/LinkTarget.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/LinkTarget.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/LinkTarget.mk Wed Nov 14 23:34:46 2012
@@ -364,7 +364,7 @@ $(call gb_LinkTarget_get_headers_target,
 $(call gb_LinkTarget_get_target,$(1)) : INCLUDE := $$(gb_LinkTarget_INCLUDE)
 $(call gb_LinkTarget_get_headers_target,$(1)) \
 $(call gb_LinkTarget_get_target,$(1)) : INCLUDE_STL := $$(gb_LinkTarget_INCLUDE_STL)
-$(call gb_LinkTarget_get_target,$(1)) : LDFLAGS := $$(gb_LinkTarget_LDFLAGS)
+$(call gb_LinkTarget_get_target,$(1)) : T_LDFLAGS := $$(gb_LinkTarget_LDFLAGS) $(LDFLAGS)
 $(call gb_LinkTarget_get_target,$(1)) : LINKED_LIBS := 
 $(call gb_LinkTarget_get_target,$(1)) : LINKED_STATIC_LIBS := 
 $(call gb_LinkTarget_get_target,$(1)) : EXTERNAL_LIBS := 
@@ -498,8 +498,12 @@ endif
 
 endef
 
+define gb_LinkTarget_add_ldflags
+$(call gb_LinkTarget_get_target,$(1)) : T_LDFLAGS += $(2)
+endef
+
 define gb_LinkTarget_set_ldflags
-$(call gb_LinkTarget_get_target,$(1)) : LDFLAGS := $(2)
+$(call gb_LinkTarget_get_target,$(1)) : T_LDFLAGS := $(2)
 endef
 
 define gb_LinkTarget_add_api

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/StaticLibrary.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/StaticLibrary.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/StaticLibrary.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/StaticLibrary.mk Wed Nov 14 23:34:46 2012
@@ -95,7 +95,9 @@ $(eval $(foreach method,\
 	add_defs \
 	set_defs \
 	set_include \
+	add_ldflags \
 	set_ldflags \
+	add_libs \
 	set_library_path_flags \
 	add_api \
 	add_linked_libs \

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/linux.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/linux.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/linux.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/linux.mk Wed Nov 14 23:34:46 2012
@@ -228,7 +228,8 @@ $(call gb_Helper_abbreviate_dirs,\
 	mkdir -p $(dir $(1)) && \
 	$(gb_CXX) \
 		$(if $(filter Library CppunitTest,$(TARGETTYPE)),$(gb_Library_TARGETTYPEFLAGS)) \
-		$(subst \d,$$,$(RPATH)) $(LDFLAGS) \
+		$(subst \d,$$,$(RPATH)) \
+		$(T_LDFLAGS) \
 		$(foreach object,$(COBJECTS),$(call gb_CObject_get_target,$(object))) \
 		$(foreach object,$(CXXOBJECTS),$(call gb_CxxObject_get_target,$(object))) \
 		$(foreach object,$(GENCXXOBJECTS),$(call gb_GenCxxObject_get_target,$(object))) \

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/macosx.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/macosx.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/macosx.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/macosx.mk Wed Nov 14 23:34:46 2012
@@ -259,14 +259,16 @@ $(call gb_Helper_abbreviate_dirs,\
 	$(PERL) $(SOLARENV)/bin/macosx-dylib-link-list.pl \
 		$(if $(filter Executable,$(TARGETTYPE)),$(gb_Executable_TARGETTYPEFLAGS)) \
 		$(if $(filter Library CppunitTest,$(TARGETTYPE)),$(gb_Library_TARGETTYPEFLAGS)) \
-		$(subst \d,$$,$(RPATH)) $(LDFLAGS) \
+		$(subst \d,$$,$(RPATH)) \
+		$(T_LDFLAGS) \
 		$(patsubst lib%.dylib,-l%,$(foreach lib,$(LINKED_LIBS),$(call gb_Library_get_filename,$(lib)))) \
 		$(patsubst %,-l%,$(EXTERNAL_LIBS)) \
 		> $${DYLIB_FILE} && \
 	$(gb_CXX) \
 		$(if $(filter Executable,$(TARGETTYPE)),$(gb_Executable_TARGETTYPEFLAGS)) \
 		$(if $(filter Library CppunitTest,$(TARGETTYPE)),$(gb_Library_TARGETTYPEFLAGS)) \
-		$(subst \d,$$,$(RPATH)) $(LDFLAGS) \
+		$(subst \d,$$,$(RPATH)) \
+		$(T_LDFLAGS) \
 		$(call gb_LinkTarget__get_liblinkflags,$(LINKED_LIBS)) \
 		$(foreach object,$(COBJECTS),$(call gb_CObject_get_target,$(object))) \
 		$(foreach object,$(CXXOBJECTS),$(call gb_CxxObject_get_target,$(object))) \

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/solaris.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/solaris.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/solaris.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/solaris.mk Wed Nov 14 23:34:46 2012
@@ -207,7 +207,8 @@ $(call gb_Helper_abbreviate_dirs,\
 	mkdir -p $(dir $(1)) && \
 	$(gb_CXX) \
 		$(if $(filter Library CppunitTest,$(TARGETTYPE)),$(gb_Library_TARGETTYPEFLAGS)) \
-		$(subst \d,$$,$(RPATH)) $(LDFLAGS) \
+		$(subst \d,$$,$(RPATH)) \
+		$(T_LDFLAGS) \
 		$(patsubst lib%.so,-l%,$(foreach lib,$(LINKED_LIBS),$(call gb_Library_get_filename,$(lib)))) \
 		$(patsubst %,-l%,$(EXTERNAL_LIBS)) \
 		$(foreach object,$(COBJECTS),$(call gb_CObject_get_target,$(object))) \

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/windows.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/windows.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/windows.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/windows.mk Wed Nov 14 23:34:46 2012
@@ -440,7 +440,7 @@ $(call gb_Helper_abbreviate_dirs_native,
 		$(if $(filter Library CppunitTest,$(TARGETTYPE)),$(gb_Library_TARGETTYPEFLAGS)) \
 		$(if $(filter StaticLibrary,$(TARGETTYPE)),$(gb_StaticLibrary_TARGETTYPEFLAGS)) \
 		$(if $(filter Executable,$(TARGETTYPE)),$(gb_Executable_TARGETTYPEFLAGS)) \
-		$(LDFLAGS) \
+		$(T_LDFLAGS) \
 		@$${RESPONSEFILE} \
 		$(foreach lib,$(LINKED_LIBS),$(call gb_Library_get_filename,$(lib))) \
 		$(patsubst %,%.lib,$(EXTERNAL_LIBS)) \

Modified: incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/winmingw.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/winmingw.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/winmingw.mk (original)
+++ incubator/ooo/branches/gbuild/main/solenv/gbuild/platform/winmingw.mk Wed Nov 14 23:34:46 2012
@@ -407,7 +407,7 @@ $(call gb_Helper_abbreviate_dirs_native,
 		$(NATIVERES) " > $${RESPONSEFILE} && \
 	$(gb_LINK) \
 		$(gb_Executable_TARGETTYPEFLAGS) \
-		$(LDFLAGS) \
+		$(T_LDFLAGS) \
 		$(gb_MINGWLIBDIR)/crt2.o \
 		$(MINGW_CLIB_DIR)/crtbegin.o \
 		@$${RESPONSEFILE} \
@@ -431,7 +431,7 @@ $(call gb_Helper_abbreviate_dirs_native,
 		$(NATIVERES) " > $${RESPONSEFILE} && \
 	$(gb_LINK) \
 		$(gb_Library_TARGETTYPEFLAGS) \
-		$(LDFLAGS) \
+		$(T_LDFLAGS) \
 		--enable-auto-image-base \
 		-e _DllMainCRTStartup@12 \
 		$(gb_MINGWLIBDIR)/dllcrt2.o \

Modified: incubator/ooo/branches/gbuild/main/sw/Library_vbaswobj.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/sw/Library_vbaswobj.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/sw/Library_vbaswobj.mk (original)
+++ incubator/ooo/branches/gbuild/main/sw/Library_vbaswobj.mk Wed Nov 14 23:34:46 2012
@@ -115,10 +115,8 @@ $(eval $(call gb_Library_add_exception_o
 ))
 
 ifeq ($(OS),LINUX)
-$(eval $(call gb_Library_set_ldflags,vbaswobj,\
-	$$(LDFLAGS) \
+$(eval $(call gb_Library_add_ldflags,vbaswobj,\
 	-Wl$(COMMA)-O1 \
-	-Wl$(COMMA)-z$(COMMA)noexecstack \
 ))
 endif
 

Modified: incubator/ooo/branches/gbuild/main/tools/Library_tl.mk
URL: http://svn.apache.org/viewvc/incubator/ooo/branches/gbuild/main/tools/Library_tl.mk?rev=1409557&r1=1409556&r2=1409557&view=diff
==============================================================================
--- incubator/ooo/branches/gbuild/main/tools/Library_tl.mk (original)
+++ incubator/ooo/branches/gbuild/main/tools/Library_tl.mk Wed Nov 14 23:34:46 2012
@@ -136,8 +136,7 @@ $(call gb_Library_use_external,tl,zlib)
 
 ifeq ($(OS),OS2)
 # YD FIXME above is not working... needs ldflags hack...
-$(eval $(call gb_Library_set_ldflags,tl,\
-    $$(LDFLAGS) \
+$(eval $(call gb_Library_add_libs,tl,\
     -lz \
 ))
 endif



